File: add_digits.py
#!/usr/bin/python3

def main():
	print ("Hello this program will add the digits of an integer until a singular digit is obtained")
	print ('For example 38 -> 3+8 = 11 -> 1+1 = 2. 2 will be returned')
	number = input('Enter an integer value: ')
	print (number + ' will reduce to ' + str(addDigits(number)))

def addDigits(num):
	sum_of_num = 0
	if len(num) == 1:
		return num 
	else: 
		for n in str(num):
			sum_of_num += int(n)
		return addDigits(str(sum_of_num))

    
if __name__ == '__main__':
    main()

File: add_digits.pl
#!/usr/bin/env perl

use strict;

print "Please enter an integer with two or more digits: ";
chomp(my $n = <STDIN>);
unless ($n =~ /^\d{2,}$/) {
    die "Bad input.\n";
}

print $n, " reduces to ", add_digits($n), "\n";

sub add_digits {
    my $n = shift;
    if (length($n) == 1) {
        return $n;
    }
    else {
        my $sum = 0;
        for my $i (split(//, $n)) {
            $sum += $i;
        }
        add_digits($sum);
    }
}
